mindmaps-graql/src/main/java/io/mindmaps/graql/internal/analytics/SumMapReduce.java

```diff
@@ -0,0 +1,91 @@
+package io.mindmaps.graql.internal.analytics;
+
+import io.mindmaps.concept.ResourceType;
+import io.mindmaps.util.Schema;
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.process.computer.KeyValue;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.util.iterator.IteratorUtils;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import static io.mindmaps.graql.internal.analytics.Analytics.getVertexType;
+
+public class SumMapReduce extends MindmapsMapReduce<Number> {
```

Comment:
All map reduce class share the same "resource type" property and type checks and essentially the same implementation for all stages, save for the actual emitting of the map and the reduce.